Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2015-04-24
Initial Package Version: 3.3.2
Upstream Status: uncertain
Origin: Khem Raj
Description: fixes build with gcc-5

Found at http://permalink.gmane.org/gmane.comp.handhelds.openembedded.core/62793

Presumably, this can be replaced by sed when gcc-5 is in the book.

diff -Naur mdadm-3.3.2.orig/bitmap.c mdadm-3.3.2/bitmap.c
--- mdadm-3.3.2.orig/bitmap.c	2014-08-21 11:16:56.000000000 +0100
+++ mdadm-3.3.2/bitmap.c	2015-04-24 04:15:15.747699137 +0100
@@ -20,7 +20,7 @@
 
 #include "mdadm.h"
 
-inline void sb_le_to_cpu(bitmap_super_t *sb)
+static inline void sb_le_to_cpu(bitmap_super_t *sb)
 {
 	sb->magic = __le32_to_cpu(sb->magic);
 	sb->version = __le32_to_cpu(sb->version);
@@ -34,7 +34,7 @@
 	sb->write_behind = __le32_to_cpu(sb->write_behind);
 }
 
-inline void sb_cpu_to_le(bitmap_super_t *sb)
+static inline void sb_cpu_to_le(bitmap_super_t *sb)
 {
 	sb_le_to_cpu(sb); /* these are really the same thing */
 }
@@ -74,7 +74,7 @@
 } bitmap_info_t;
 
 /* count the dirty bits in the first num_bits of byte */
-inline int count_dirty_bits_byte(char byte, int num_bits)
+static inline int count_dirty_bits_byte(char byte, int num_bits)
 {
 	int num = 0;
 
